hp/src/ErrorHandler.hs

```diff
@@ -0,0 +1,11 @@
+module ErrorHandler where
+
+import AbstractSyntaxTree (Identifier)
+
+data SemanticError
+    = UndeclaredVariable Identifier
```

Comment:
It is working fine for the variables inside the do-notation, but not when you want to use that variable as a parameter:
![Screenshot from 2024-07-04 21-27-41](https://github.com/luiggymamani/hufflepuff-own-language/assets/135136456/b20acd07-a0f1-490f-94c6-48b328d23cb5)

and also when you want to use a parameter that was not declared:
![2024-07-04_21-38](https://github.com/luiggymamani/hufflepuff-own-language/assets/135136456/8d46579b-fc0b-4133-af65-5f9bad8c6383)